Essentials/src/com/earth2me/essentials/signs/SignDisposal.java

```diff
@@ -11,9 +12,23 @@ public SignDisposal() {
         super("Disposal");
     }
 
+    @Override
+    protected boolean onSignCreate(ISign sign, User player, String username, IEssentials ess) throws SignException, ChargeException {
+        if (!player.isAuthorized("essentials.signs.disposal.name")) {
+            sign.setLine(1, "");
+            sign.setLine(2, "");
+            sign.setLine(3, "");
+        }
+        return true;
+    }
+
     @Override
     protected boolean onSignInteract(final ISign sign, final User player, final String username, final IEssentials ess) {
-        player.getBase().openInventory(ess.getServer().createInventory(player.getBase(), 36, tl("disposal")));
+        String title = sign.getLine(1) + " " + sign.getLine(2) + " " + sign.getLine(3);
+        if (title.trim().isEmpty()) {
+            title = tl("disposal");
+        }
+        player.getBase().openInventory(ess.getServer().createInventory(player.getBase(), 36, title.trim()));
```

Comment:
This is a *very* subtle breaking change, but if server admins intentionally add leading spaces to the `disposal` locale string, `title.trim()` will remove them and break the layout. I'd prefer trimming the title only if it was set on the sign. Granted, I'm not sure if anyone really does this, but on the off chance people do we'll be breaking their disposal signs.

Reply:
Changes applied :P